### ISSUE TITLE
Script arguments

### ISSUE THREAD

**Issue by hchuphal** [hasanferit/DeepFault#2]
Hej,

Could you please provide some examples of as to how to run the scripts ´?
python2.7 run.py
Using TensorFlow backend.
usage: run.py [-h] [-V] -M MODEL [-DS {mnist,cifar10}] [-A APPROACH]
              [-D DISTANCE] [-C CLASS] [-AC ACTIVATION] [-SN SUSPICIOUS_NUM]
              [-SS STEP_SIZE] [-R REPEAT] [-S SEED] [-ST STAR] [-LOG LOGFILE]

thanks!

**Comment by hfeniser** [comment]
model_name     =>  Name of the -Keras- model file. Note that architecture file (i.e. json) and weights file should be saved separately. If you already trained and saved your model into one file you might want to change the corresponding "load_model" function accordingly.
    dataset        =   Name of the dataset to be used. Current implementation supports only 'mnist' and 'cifar'. However, adding another is not difficult.
    selected_class =   In DeepFault, we find the suspicious neurons for each class separately. This argument is a number between 0 and 9 for mist or cifar.
    step_size      =   We multiply the gradient values by this parameter for scaling up or down the change while synthesizing a new input.
    distance       =   The maximum amount of distance (l_inf norm) between the original and the synthesized input.
    approach       =   The approach for finding the suspicious neurons. Current implementation supports 'tarantula', 'ochiai', 'dstar' and 'random'. Note that random can be selected only after running all others.
    susp_num       =   Number of neurons considered suspicious.
    repeat         =   This added to repeat the experiments to reduce the randomness effect. Useless at the moment.
    seed           =   Seed for the random process. Added for reproducibility.
    star           =   This corresponds to the "star" parameter of dstar approach.
    logfile_name   =   Name of the file that the results to be saved.

According to this, an example usage of DeepFault can be:

python run.py --model path_to_your_mnist_model_file --dataset mnist --selected_class 0 --approach tarantula --suspicious_num 10

**Comment by tangjin960322** [comment]
Sorry,I haven't understand how to run this program,could you describe it clearly?

**Comment by hfeniser** [comment]
Which part is not clear exactly? You might want to check the [paper](https://arxiv.org/abs/1902.05974) first.

**Comment by gerasimou** [comment]
Hi @tangjin960322,

In addition to what suggest by @hasanferit, you may want to check the comment above
https://github.com/hasanferit/DeepFault/issues/2#issuecomment-587407713
which explains command line settings and flags for running the Python scripts.

Please let us know if anything is unclear.

Regards,
Simos

**Comment by tangjin960322** [comment]
Thank you very much,your answer is timely for my experiment.You can call me Jin(it's my name)




At 2020-10-09 17:33:03, "Simos Gerasimou" <notifications@github.com> wrote:

Hi @tangjin960322,

In addition to what suggest by @hasanferit, you may want to check the comment above
#2 (comment)
which explains command line settings and flags for running the Python scripts.

Please let us know if anything is unclear.

Regards,
Simos

**Comment by tangjin960322** [comment]
非常抱歉，一而再再而三的打扰您！（PS：我的英文很差，我尽可能使用我会的英文来写，不能够表达清楚地，我再使用中文，请您见谅）
sorry,I think i have some questions about the running of this program:
1、model path_to_your_mnist_model_file、dataset mnist、selected_class 0 、approach tarantula、suspicious_num 10：这四个元素分别指什么呢？
这里边的dataset mnist是我已经下载下来的，具体图片如下：
但是，很抱歉，我不会使用，因此能否麻烦您再给我一些指导？















At 2020-10-09 17:33:03, "Simos Gerasimou" <notifications@github.com> wrote:

Hi @tangjin960322,

In addition to what suggest by @hasanferit, you may want to check the comment above
#2 (comment)
which explains command line settings and flags for running the Python scripts.

Please let us know if anything is unclear.

Regards,
Simos
